### PR TITLE
Added tags and tag list pages

### PR DESCRIPTION
**Pull request by PaulMorel1**
# Added Tags and Tag List Pages

Added tagging and tag list page, which is a list of posts with a given tag.

1. Added dedicated tag model.
2. Added various tag components.
3. Added tagList template for the list of posts with a given tag. I also added the routing to make the list pages dynamically.
4. Added a list of tags to the footer on a blog post. These link out to the tag list pages.

This work is basically a copy of the existing Authors work. I took the Authors work as idiomatic, and worked from there.

## Screenshots

tag-list-page

tag-form

tag-post

## Why should we merge this?

1. This doesn't introduce any new patterns.
2. Tags have to be a full fledged model in order for them to work, within the existing framework, as tags commonly work on blogs.

In short, this is a good starting point for tags. It is extensible, and it uses existing patterns rather than adding new ones.

## Future Work

1. Could make a list page for ALL tags on the blog platform. This is easy enough, but it's fine to leave out of the initial tags work.
2. Could add an ad-hoc form for creating new tags on a post, rather than picking from a list. This is a good improvement, but this would be a new pattern, so I thought it best to leave this as an improvement for now.

### CODE REVIEW

**Comment by PaulMorel1** [comment]
@DirtyF @spbyrne I'd love a review. :D

**Comment by spbyrne** [comment]
This is awesome! Sorry, I haven't been actively maintaining this project but I greatly appreciate improvements like this.
